Remove Document::setDesignMode overload

rwlbuis opened the pull request:
#### 0ddbfa3ca8f34f49b95eeacd8b171e1caab1c148
<pre>
Remove Document::setDesignMode overload
<a href="https://bugs.webkit.org/show_bug.cgi?id=259389">https://bugs.webkit.org/show_bug.cgi?id=259389</a>

Reviewed by Darin Adler.

Remove the setDesignMode that is not actually used.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::setDesignMode):
* Source/WebCore/dom/Document.h:

Canonical link: <a href="https://commits.webkit.org/266285@main">https://commits.webkit.org/266285@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d84f1f12245190446412c08ae89dd007eeb26762

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13363 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13677 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14009 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15100 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12723 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13442 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16184 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13703 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15403 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13530 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14182 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11303 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/15741 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11470 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12058 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/19108 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12545 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12225 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15438 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12730 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10588 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12000 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3275 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16323 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12571 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -2039,7 +2039,6 @@ Ref<TreeWalker> Document::createTreeWalker(Node& root, unsigned long whatToShow,
 void Document::scheduleFullStyleRebuild()
 {
     m_needsFullStyleRebuild = true;
-    scheduleStyleRecalc();
```
Why are we making this change??

rniwa commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -2180,9 +2179,6 @@ void Document::resolveStyle(ResolveStyleType type)
         updatedCompositingLayers = frameView.updateCompositingLayersAfterStyleChange();
 
-        if (m_renderView->needsLayout())
-            frameView.layoutContext().scheduleLayout();
```
Ditto.

rwlbuis committed:
Remove Document::setDesignMode overload
https://bugs.webkit.org/show_bug.cgi?id=259389

Reviewed by Darin Adler.

Remove the setDesignMode that is not actually used.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::setDesignMode):
* Source/WebCore/dom/Document.h:

Canonical link: https://commits.webkit.org/266285@main
